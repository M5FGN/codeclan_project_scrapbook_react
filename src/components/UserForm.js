import React from "react";
import { useHistory } from "react-router-dom";
import Logo from "./Logo";
import Footer from "../components/Footer";

const UserForm = ({ setUser }) => {
  let history = useHistory();

  const handleLogIn = function (event) {
    event.preventDefault();
    const name = event.target.name.value;
    const password = event.target.password.value;
    console.log(name, password);

    fetch(
      `http://localhost:8080/api/users/login?name=${name}&password=${password}`
    )
      .then((res) => res.json())
      .then((user) => {
        setUser(user);
        window.sessionStorage.setItem("userId", user.id);
        console.log(window.sessionStorage.getItem("userId"));
        history.push("/");
      });
  };

  return (
    <div>

      <Logo />

      <div className="card">
        <h2>Log In to ScrapBook ...</h2>
      <form className="form" onSubmit={handleLogIn}>
        <label>Username ... </label>
        <input
          type="text"
          placeholder="Enter User Name ..."
          name="name"
        ></input>
        <label>Password ...</label>
        <input
          type="text"
          placeholder="Enter Password ..."
          name="password"
        ></input>
        <input className="button" type="submit" value="Log In ..."></input>
      </form>
      <h3>
        <a href="/signup">Don't have an account?</a>
      </h3>
    </div>
    <Footer />
    </div>
  );
};
export default UserForm;
