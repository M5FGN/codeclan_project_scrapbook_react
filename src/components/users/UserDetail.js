import React from 'react';

const UserDetail = () => {

    return (
        <div>
            <h1>User Name</h1>
            <p> Profile Pic</p>
            <p>Age</p>
            <p>Bio</p>
        </div>

    )

}
export default UserDetail;