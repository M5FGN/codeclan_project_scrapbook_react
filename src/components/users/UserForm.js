import React from 'react';

const UserForm = () => {

return (
<div>
<h1>User Form</h1>
<form>
    <label>Username ... </label>
<input type="text" placeholder="Enter User Name ..." name = "name"></input>
<label>Password ...</label>
<input type="text" placeholder="Enter Password ..." name = "password"></input>
<input type="submit" value= "Login"></input>
</form>

</div>
)
}
export default UserForm;