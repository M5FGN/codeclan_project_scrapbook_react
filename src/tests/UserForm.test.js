import React from 'react';
import UserForm from '../components/UserForm';
import UserDetail from '../components/users/UserDetail'
import { render, fireEvent } from '@testing-library/react';


describe("UserForm", () => {
    let container;
   
    const mockChangeValue = jest.fn();
    const stubbedUser = {
      name: "Ahmet",
      password: "metallica"
    };

  
    // first test case to check all fields are empty
    it("shows all required input fields with empty values", () => {
      const { getByTestId } = render(
        <UserForm setUser={stubbedUser} />
      );
  
      expect(getByTestId("user-name").value).toBe("");
      expect(getByTestId("password").value).toBe("");
    });

    // second test case to check user can login
    it("login test user", () => {
        const { getByTestId } = render(
          <UserForm setUser={stubbedUser} />
        );
        getByTestId("user-name").value = "Ahmet";
        getByTestId("password").value = "metallica";

        getByTestId("log-in").onsubmit;

        let container = render(<UserDetail user={stubbedUser} />);
        expect(container.getByTestId('name')).toHaveTextContent("Ahmet");
    
      });
    })
