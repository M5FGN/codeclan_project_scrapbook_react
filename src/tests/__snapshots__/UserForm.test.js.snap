// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <img
      alt=""
      class="logo"
      src="logo.jpg"
    />
    <div
      class="card formbox"
    >
      <h2>
        Log In to ScrapBook ...
      </h2>
      <form
        data-testid="logged-in"
      >
        <label>
          Username:
        </label>
        <br />
        <input
          data-testid="user-name"
          id="name"
          placeholder="Enter User Name ..."
          type="text"
        />
        <br />
        <label>
          Password:
        </label>
        <br />
        <input
          data-testid="password"
          id="password"
          placeholder="Enter Password ..."
          type="text"
        />
        <br />
        <input
          class="button"
          data-testid="log-in"
          type="submit"
          value="Log In ..."
        />
      </form>
      <h3>
        <a
          href="/signup"
        >
          Don't have an account?
        </a>
      </h3>
    </div>
    <div>
      <p
        class="center"
      >
        Copyright Â© 2021 - Ahmet, Liam, Mary, Pooja
      </p>
    </div>
  </div>
</DocumentFragment>
`;
